Incorrect class name is used when the generated test class requires a number to be added (e.g. Client.class in EE 10 but was split into Client1/2/3/4 in EE 11)

Example from jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/entityManager/Client2AppmanagedTest.java:

```
public class Client2AppmanagedTest extends ee.jakarta.tck.persistence.core.entityManager.Client2 {
    static final String VEHICLE_ARCHIVE = "jpa_core_entityManager_appmanaged_vehicle";
...
        @TargetsContainer("tck-appclient")
        @OverProtocol("appclient")
        @Deployment(name = VEHICLE_ARCHIVE, order = 2)
        public static EnterpriseArchive createDeploymentVehicle(@ArquillianResource TestArchiveProcessor archiveProcessor) {
        // Client
            // the jar with the correct archive name
            JavaArchive jpa_core_entityManager_appmanaged_vehicle_client = ShrinkWrap.create(JavaArchive.class, "jpa_core_entityManager_appmanaged_vehicle_client.jar");
            // The class files
            jpa_core_entityManager_appmanaged_vehicle_client.addClasses(
            com.sun.ts.tests.common.vehicle.VehicleRunnerFactory.class,
            com.sun.ts.tests.common.vehicle.ejb3share.UseEntityManager.class,
            com.sun.ts.tests.common.vehicle.ejb3share.EJB3ShareIF.class,
            com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleIF.class,
            com.sun.ts.tests.common.vehicle.ejb3share.UseEntityManagerFactory.class,
            com.sun.ts.tests.common.vehicle.EmptyVehicleRunner.class,
            ee.jakarta.tck.persistence.common.PMClientBase.class,
            com.sun.ts.tests.common.vehicle.VehicleRunnable.class,
            com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleRunner.class,
            com.sun.ts.tests.common.vehicle.ejb3share.UserTransactionWrapper.class,
            com.sun.ts.lib.harness.EETest.class,
            com.sun.ts.lib.harness.ServiceEETest.class,
            com.sun.ts.tests.common.vehicle.ejb3share.EntityTransactionWrapper.class,
            com.sun.ts.tests.common.vehicle.VehicleClient.class,
            com.sun.ts.tests.common.vehicle.ejb3share.NoopTransactionWrapper.class
            );


```

I'm not sure but with this change + https://github.com/scottmarlow/jakartaee-tck-tools/tree/jakartaee-tck-tools-pull-95, I am getting the following compile failure (without my hack change that I mentioned earlier):

> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3PmservletTest.java:[79,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientPmservletTest.java:[70,59] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientAppmanagednotxTest.java:[110,63] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3AppmanagedTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientAppmanagedTest.java:[110,63] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3PuservletTest.java:[79,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientPuservletTest.java:[71,59] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientStateful3Test.java:[109,63] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client5Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client2PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3Stateful3Test.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6PmservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3AppmanagednotxTest.java:[118,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1Stateful3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client3Stateless3Test.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client1PuservletTest.java:[80,71] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/jpa22/query/stream/ClientStateless3Test.java:[109,63] package com.sun.ts.tests.jpa.jpa22.query.stream.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client7Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6Stateless3Test.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client6AppmanagednotxTest.java:[119,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> [ERROR] platformtck/jpa/platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4AppmanagedTest.java:[120,75] package com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client does not exist
> 


From platform-tests/src/main/java/ee/jakarta/tck/persistence/core/criteriaapi/CriteriaQuery/Client4AppmanagedTest.java:
```
        // Ejb
            // the jar with the correct archive name
            JavaArchive jpa_core_criteriaapi_CriteriaQuery_appmanaged_vehicle_ejb = ShrinkWrap.create(JavaArchive.class, "jpa_core_criteriaapi_CriteriaQuery_appmanaged_vehicle_ejb.jar");
            // The class files
            jpa_core_criteriaapi_CriteriaQuery_appmanaged_vehicle_ejb.addClasses(
                com.sun.ts.tests.common.vehicle.ejb3share.EJB3ShareBaseBean.class,
                com.sun.ts.tests.common.vehicle.VehicleRunnerFactory.class,
                com.sun.ts.tests.common.vehicle.ejb3share.UseEntityManager.class,
                com.sun.ts.tests.common.vehicle.ejb3share.EJB3ShareIF.class,
                com.sun.ts.lib.harness.EETest.Fault.class,
                com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleIF.class,
                com.sun.ts.tests.common.vehicle.ejb3share.UseEntityManagerFactory.class,
                ee.jakarta.tck.persistence.common.PMClientBase.class,
                ee.jakarta.tck.persistence.common.schema30.Util.class,
                com.sun.ts.tests.common.vehicle.VehicleRunnable.class,
                com.sun.ts.tests.common.vehicle.appmanaged.AppManagedVehicleBean.class,
                com.sun.ts.tests.common.vehicle.ejb3share.UserTransactionWrapper.class,
                ee.jakarta.tck.persistence.core.criteriaapi.CriteriaQuery.Client4.class,
                com.sun.ts.lib.harness.EETest.class,
                com.sun.ts.lib.harness.ServiceEETest.class,
                com.sun.ts.tests.common.vehicle.ejb3share.EntityTransactionWrapper.class,
                com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client.ExpectedResult.class,
                com.sun.ts.lib.harness.EETest.SetupException.class,
                com.sun.ts.tests.common.vehicle.VehicleClient.class,
                com.sun.ts.tests.common.vehicle.ejb3share.NoopTransactionWrapper.class
            );

```

It looks like com.sun.ts.tests.jpa.core.criteriaapi.CriteriaQuery.Client.ExpectedResult.class` was for nested Client.ExpectedResult class in the EE 10 [Client.java](https://github.com/jakartaee/platform-tck/blob/10.0.x/src/com/sun/ts/tests/jpa/core/criteriaapi/CriteriaQuery/Client.java)

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Lib.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Lib.java
@@ -13,7 +13,7 @@ public class Lib {
     EE11toEE10Mapping mapping;
 
     public Lib() {
-        this(new DefaultEEMapping());
+        this(DefaultEEMapping.getInstance());
     }
     public Lib(EE11toEE10Mapping mapping) {
         this.mapping = mapping;

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/PackageTarget.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/PackageTarget.java
@@ -43,7 +43,7 @@ public class PackageTarget {
     EE11toEE10Mapping mapping;
 
     public PackageTarget(ProjectWrapper project, Target pkgTarget) {
-        this(project, pkgTarget, new DefaultEEMapping());
+        this(project, pkgTarget, DefaultEEMapping.getInstance());
     }
     public PackageTarget(ProjectWrapper project, Target pkgTarget, EE11toEE10Mapping mapping) {
         this.project = project;

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Utils.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Utils.java
@@ -253,10 +253,12 @@ public class Utils {
                     }
                     int dollar = dotClass.indexOf('$');
                     if(dollar > 0) {
+                        // This is an inner class, if it is anonymous, add it to the list
                         if(Character.isDigit(dotClass.charAt(dollar+1))) {
                             anonymousClasses.add(dotClass);
                             continue;
                         } else {
+                            // Otherwise, create a literal class name for the inner class
                             clazz = dotClass.replace('$', '.');
                         }
                     }

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Vehicles.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/Vehicles.java
@@ -63,7 +63,7 @@ public class Vehicles {
     private TsFileSet jarElements;
     private List<Lib> earLibs = new ArrayList<>();
     private List<Lib> warLibs = new ArrayList<>();
-    private EE11toEE10Mapping mapping = new DefaultEEMapping();
+    private EE11toEE10Mapping mapping = DefaultEEMapping.getInstance();
 
 
     public Vehicles(final String name, final String vehiclePkgDir, AttributeMap attributes, RuntimeConfigurable rc, Location location) {

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/DefaultEEMapping.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/DefaultEEMapping.java
@@ -1,6 +1,9 @@
 package tck.jakarta.platform.ant.api;
 
+import java.nio.file.Path;
+
 public class DefaultEEMapping implements EE11toEE10Mapping {
+    private static final DefaultEEMapping INSTANCE = new DefaultEEMapping();
     // Mappings from EE11 to EE10 package prefixes
     private static final String[] EE11_PKG_PREFIXES = {
             "ee.jakarta.tck.persistence.common.schema30", "com.sun.ts.tests.jpa.common.schema30",
@@ -9,6 +12,39 @@ public class DefaultEEMapping implements EE11toEE10Mapping {
             "ee.jakarta.tck.persistence.entitytest.persist.oneXmanyFetchEager","com.sun.ts.tests.jpa.core.entitytest.persist.oneXmanyFetchEager",
             "ee.jakarta.tck.persistence", "com.sun.ts.tests.jpa"
     };
+    //  A mapping from EE11 to EE10 class names
+    private String[] testClassMappings = {"", ""};
+
+    public static DefaultEEMapping getInstance() {
+        return INSTANCE;
+    }
+    private DefaultEEMapping() {}
+
+    @Override
+    public String addTestClassMapping(Class<?> ee11Class, Path tsHome) {
+        String ee11Name = ee11Class.getName();
+        String ee10Name = null;
+        char lastChar = ee11Name.charAt(ee11Name.length() - 1);
+        if (Character.isDigit(lastChar)) {
+            String simpleNameN = ee11Class.getSimpleName();
+            String simpleName = simpleNameN.substring(0, simpleNameN.length() - 1);
+
+            // See if this class exists in EE10
+            String ee10Pkg = ee11Class.getPackageName();
+            ee10Pkg = getEE10TestPackageName(ee10Pkg);
+            String ee10Path = ee10Pkg.replace('.', '/');
+            Path ee10TestPathN = tsHome.resolve("classes").resolve(ee10Path).resolve(simpleNameN + ".class");
+            Path ee10TestPath = tsHome.resolve("classes").resolve(ee10Path).resolve(simpleName + ".class");
+            if (!ee10TestPathN.toFile().exists() && ee10TestPath.toFile().exists()) {
+                // Add the mapping
+                ee10Name = ee10Pkg + "." + simpleName + ".class";
+                testClassMappings[0] = ee11Name + ".class";
+                testClassMappings[1] = ee10Name;
+            }
+        }
+        return ee10Name;
+    }
+
     @Override
     public String getEE10TestPackageName(String ee11Name) {
         String ee10Name = ee11Name;
@@ -24,6 +60,9 @@ public class DefaultEEMapping implements EE11toEE10Mapping {
     @Override
     public String getEE11Name(String ee10Name) {
         String ee11Name = ee10Name;
+        if(ee10Name.equals(testClassMappings[1])) {
+            return testClassMappings[0];
+        }
         for (int i = 1; i < EE11_PKG_PREFIXES.length; i += 2) {
             if (ee11Name.startsWith(EE11_PKG_PREFIXES[i])) {
                 ee11Name = ee10Name.replace(EE11_PKG_PREFIXES[i], EE11_PKG_PREFIXES[i-1]);

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/EE11toEE10Mapping.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/EE11toEE10Mapping.java
@@ -1,9 +1,17 @@
 package tck.jakarta.platform.ant.api;
 
+import java.nio.file.Path;
+
 /**
  * Maps EE11 names to EE10 names.
  */
 public interface EE11toEE10Mapping {
+    /**
+     * This looks at the EE11 class and if it sends in a number and does not exist in EE10, a mapping from
+     * the com.sun.ts.tests.x.Class.class to the com.sun.ts.tests.x.ClassN.class is added.
+     */
+    public String addTestClassMapping(Class<?> ee11Class, Path tsHome);
+
     /**
      * Map a package name from EE11 to EE10 to determine the location of the build.xml file.
      * The returned name is relative to the EE10 platform TCK com.sun.ts.tests package under

--- a/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/TestPackageInfoBuilder.java
+++ b/tools/tck-rewrite-ant/src/main/java/tck/jakarta/platform/ant/api/TestPackageInfoBuilder.java
@@ -109,7 +109,7 @@ public class TestPackageInfoBuilder {
             testMethodInfos.add(new TestMethodInfo(testMethod, Exception.class.getSimpleName()));
         }
 
-        return buildTestPackgeInfoEx(clazz, testMethodInfos, new DefaultEEMapping());
+        return buildTestPackgeInfoEx(clazz, testMethodInfos, DefaultEEMapping.getInstance());
     }
     public TestPackageInfo buildTestPackgeInfoEx(Class<?> clazz, List<TestMethodInfo> testMethods, EE11toEE10Mapping mapping) throws IOException {
         TestPackageInfo testPackageInfo = new TestPackageInfo(clazz, testMethods, mapping);
@@ -136,7 +136,7 @@ public class TestPackageInfoBuilder {
         for (String testMethod : testMethods) {
             testMethodInfos.add(new TestMethodInfo(testMethod, Exception.class.getSimpleName()));
         }
-        return buildTestClientsEx(clazz, testMethodInfos, new DefaultEEMapping());
+        return buildTestClientsEx(clazz, testMethodInfos, DefaultEEMapping.getInstance());
     }
 
     /**
@@ -151,6 +151,8 @@ public class TestPackageInfoBuilder {
      */
     public List<TestClientInfo> buildTestClientsEx(Class<?> clazz, List<TestMethodInfo> testMethods, EE11toEE10Mapping mapping) throws IOException {
         ArrayList<TestClientInfo> testClientInfos = new ArrayList<>();
+        // Add the test class mapping
+        mapping.addTestClassMapping(clazz, tsHome);
         // The simple name, e.g., MyTest for com.sun.*.MyTest
         String testClassSimpleName = clazz.getSimpleName();
         String pkg = clazz.getPackageName();

--- a/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DefaultMappingsTest.java
+++ b/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DefaultMappingsTest.java
@@ -1,0 +1,32 @@
+package tck.conversion.ant.api;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import tck.jakarta.platform.ant.TsFileSet;
+import tck.jakarta.platform.ant.Utils;
+import tck.jakarta.platform.ant.api.DefaultEEMapping;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DefaultMappingsTest {
+    static Path tsHome = Paths.get(System.getProperty("ts.home"));
+
+    @EnabledIfSystemProperty(named = "ts.home", matches = ".*")
+    @Test
+    public void testClassNMappings() {
+        DefaultEEMapping mappings = (DefaultEEMapping) DefaultEEMapping.getInstance();
+        Class<?> baseTestClass = ee.jakarta.tck.persistence.core.entityManager.Client2.class;
+        String ee10Name = mappings.addTestClassMapping(baseTestClass, tsHome);
+        Assertions.assertEquals("com.sun.ts.tests.jpa.core.entityManager.Client.class", ee10Name);
+
+        // TsFileSet(String dir, String prefix, List<String> includes) {
+        Path testDir = tsHome.resolve("classes");
+        TsFileSet fileSet = new TsFileSet(testDir.toString(), "", List.of("com/sun/ts/tests/jpa/core/entityManager/Client.class"));
+        String classes = Utils.getClassFilesString(mappings, List.of(fileSet), new ArrayList<>());
+        Assertions.assertEquals(baseTestClass.getName()+".class", classes);
+    }
+}

--- a/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DeploymentMethodTest.java
+++ b/tools/tck-rewrite-ant/src/test/java/tck/conversion/ant/api/DeploymentMethodTest.java
@@ -57,7 +57,7 @@ public class DeploymentMethodTest {
         List<TestMethodInfo> testMethods = Arrays.asList(new TestMethodInfo("bytesMsgNullStreamTopicTest", "Exception"),
                 new TestMethodInfo("bytesMessageTopicTestsFullMsg", "Exception"),
                 new TestMethodInfo("bytesMessageTNotWriteable", "Exception"));
-        TestPackageInfo pkgInfo = builder.buildTestPackgeInfoEx(BytesMsgTopicTests.class, testMethods, new DefaultEEMapping());
+        TestPackageInfo pkgInfo = builder.buildTestPackgeInfoEx(BytesMsgTopicTests.class, testMethods, DefaultEEMapping.getInstance());
         System.out.println(pkgInfo);
 
         System.out.println("TestClasses:");
@@ -86,7 +86,7 @@ public class DeploymentMethodTest {
         TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
         List<String> testMethods = Arrays.asList("bytesMsgNullStreamTopicTest", "bytesMessageTopicTestsFullMsg", "bytesMessageTNotWriteable");
         List<TestMethodInfo> testMethodsEx = toExMethods(testMethods);
-        List<TestClientInfo> testClientInfos = builder.buildTestClientsEx(BytesMsgTopicTests.class, testMethodsEx, new DefaultEEMapping());
+        List<TestClientInfo> testClientInfos = builder.buildTestClientsEx(BytesMsgTopicTests.class, testMethodsEx, DefaultEEMapping.getInstance());
         for (TestClientInfo clientInfo : testClientInfos) {
             System.out.println(clientInfo);
         }
@@ -177,7 +177,7 @@ public class DeploymentMethodTest {
         TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
         List<String> testMethods = Arrays.asList("initOrder", "appName");
         Class<?> baseTestClass = com.sun.ts.tests.ejb30.assembly.initorder.warejb.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -192,7 +192,7 @@ public class DeploymentMethodTest {
         TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
         List<String> testMethods = Arrays.asList("testModernConnector");
         Class<?> baseTestClass = com.sun.ts.tests.ejb32.mdb.modernconnector.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -206,7 +206,7 @@ public class DeploymentMethodTest {
         TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
         List<String> testMethods = Arrays.asList("checkEnvEntry", "testDTO");
         Class<?> baseTestClass = com.sun.ts.tests.ejb30.misc.sameejbclass.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -220,7 +220,7 @@ public class DeploymentMethodTest {
         TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
         List<String> testMethods = Arrays.asList("test1");
         Class<?> baseTestClass = com.sun.ts.tests.ejb30.zombie.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -261,7 +261,7 @@ public class DeploymentMethodTest {
                 "secondLevelJarEJB", "secondLevelDir", "secondLevelDirEJB", "postConstructInvokedInSuperElseWhere",
                 "remoteAdd", "remoteAddByHelloEJB", "remoteAddByHelloEJBFromAssemblyBean");
         Class<?> baseTestClass = com.sun.ts.tests.ejb30.assembly.librarydirectory.custom.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -276,7 +276,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("test5", "com.sun.ts.lib.harness.EETest.Fault")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb.ee.tx.session.stateless.bm.Tx_Multi.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -307,7 +307,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("testContextSetupCompleted", "Exception")
         );
         Class<?> baseTestClass = com.sun.ts.tests.connector.localTx.msginflow.MDBClient.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -328,7 +328,7 @@ public class DeploymentMethodTest {
                 "joinTransactionTransactionRequiredExceptionTest"
                 );
         Class<?> baseTestClass = ee.jakarta.tck.persistence.ee.entityManager.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, toExMethods(testMethods), DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -341,7 +341,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("writeLockTimeout", "")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb32.lite.timer.basic.concurrency.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -357,7 +357,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("accessTimersSingleton", "")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb32.lite.timer.basic.sharing.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -369,7 +369,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("noAnnotationTest", "com.sun.ts.tests.ejb30.common.helper.TestFailedException")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb32.relaxedclientview.singleton.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -381,7 +381,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("noAnnotationTest", "com.sun.ts.tests.ejb30.common.helper.TestFailedException")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb32.relaxedclientview.stateful.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -396,7 +396,7 @@ public class DeploymentMethodTest {
 
         );
         Class<?> baseTestClass = ee.jakarta.tck.persistence.core.callback.listener.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -412,7 +412,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("libSubdirNotScanned", "Exception")
         );
         Class<?> baseTestClass = com.sun.ts.tests.ejb30.assembly.metainfandlibdir.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -424,7 +424,7 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("testConnectionEventListener", "Exception")
         );
         Class<?> baseTestClass = com.sun.ts.tests.connector.xa.event.eventClient1.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }
@@ -442,7 +442,19 @@ public class DeploymentMethodTest {
                 new TestMethodInfo("test7", "Exception")
         );
         Class<?> baseTestClass = ee.jakarta.tck.persistence.ee.propagation.cm.extended.Client.class;
-        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, new DefaultEEMapping());
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
+        System.out.println(packageInfo);
+        System.out.println(packageInfo.getTestClientFiles());
+    }
+
+    @Test
+    public void test_jpa_core_entityManager() throws IOException {
+        TestPackageInfoBuilder builder = new TestPackageInfoBuilder(tsHome);
+        List<TestMethodInfo> testMethods = Arrays.asList(
+                new TestMethodInfo("persistExceptionsTest", "Exception")
+        );
+        Class<?> baseTestClass = ee.jakarta.tck.persistence.core.entityManager.Client2.class;
+        TestPackageInfo packageInfo = builder.buildTestPackgeInfoEx(baseTestClass, testMethods, DefaultEEMapping.getInstance());
         System.out.println(packageInfo);
         System.out.println(packageInfo.getTestClientFiles());
     }

